### PR TITLE
Use new UseServiceProviderFactory overloads, clean up samples

### PR DESCRIPTION
**Pull request by Tratcher**
This wraps up https://github.com/aspnet/Extensions/issues/612 by consuming the new UseServiceProviderFactory in the generic web host adapter.

I also cleaned up the samples.

### CODE REVIEW

File: src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj [hunk 1]
```diff
@@ -22,6 +22,7 @@
     <Reference Include="Microsoft.Extensions.Configuration" />
     <Reference Include="Microsoft.Extensions.DependencyInjection" />
     <Reference Include="Microsoft.Extensions.FileProviders.Physical" />
+    <Reference Include="Microsoft.Extensions.Hosting" />
```
Comment (davidfowl) [inline src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj]: What’s this for?

Comment (Tratcher) [reply src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj]: The UseDefaultServiceProvider extension is in there.

Comment (davidfowl) [reply src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj]: Ugh, I don’t like those dependencies here just for that method.

Comment (Tratcher) [reply src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj]: I can hoist the code...

Comment (davidfowl) [reply src/Hosting/Hosting/src/Microsoft.AspNetCore.Hosting.csproj]: Yea, hoist it

**Review by davidfowl — APPROVED** [review]

**Pull request merged**